mpqp/core/instruction/measurement/basis_measure.py

```diff
@@ -79,30 +88,32 @@ def to_other_language(
         if language == Language.QISKIT:
             from qiskit.circuit import Measure
 
-            if isinstance(self.basis, ComputationalBasis):
-                return Measure()
-            else:
-                raise NotImplementedError(f"{type(self.basis)} not handled")
+            return Measure()
         else:
             raise NotImplementedError(f"{language} is not supported")
 
+    @property
+    def pre_measure(self):
+        return self.basis.to_computational()
+
     def __repr__(self) -> str:
         targets = (
             f"{self.targets}" if (not self._dynamic and len(self.targets)) != 0 else ""
         )
         options = ""
         if self.shots != 1024:
-            options += f", shots={self.shots}" if targets else f"shots={self.shots}"
+            options += f"shots={self.shots}"
         if not isinstance(self.basis, ComputationalBasis):
             options += (
                 f", basis={self.basis}"
-                if targets and options
-                else f"basis={self.shots}"
+                if len(options) != 0 or len(targets) != 0
+                else f"basis={self.basis}"
             )
         if self.label is not None:
             options += (
                 f", label={self.label}"
-                if targets and options
-                else f"label={self.shots}"
+                if len(options) != 0 or len(targets) != 0
+                else f"label={self.label}"
```

Comment:
It's a tricky bit of logic: the chore part of it is: we should add a coma if things were written before. So to do that, at each step we have to check if there are things before or not.
Maybe an example can be useful to understand this ? let's say `target == []` but `basis != ComputationalBasis` and `label != ""`. In that case the first if will make it so there is no comma before the `basis`, but one is added before the `label`. Now, you can play with the different possible combinations, to see that the two `if` are indeed required. (We could even have a separate function that does something like `add_comma_if_needed`, but it would be so niche that I don't think it's worth it)

Reply:
Ok my bad, my mistake comes from the fact that I missed the coma while I was reading. Everything makes sense. I resolve both comments